### ISSUE TITLE
Possible to inline SVG?

### ISSUE THREAD

**Issue by rolandliwag** [assetgraph/assetgraph-builder#166]
I'd like to be able to style SVG icons with CSS but to maintain a separate asset. Is it possible to reference an SVG image in my HTML and then have it inlined with the build? Also is there a way to make this work during development as well?

eg:
```
   <a src="/path/to/icon.svg"></a>
```

would build to:
```
   <a><svg><path ... /></svg></a>
```

**Comment by Munter** [comment]
How would you style it?

**Comment by rolandliwag** [comment]
Something like this:

```
a svg {
    fill: #ff0000;
}
```

**Comment by slang800** [comment]
You'd probably be better off using [svg symbols](https://css-tricks.com/svg-symbol-good-choice-icons/)... otherwise you're going to be duplicating the icons a lot. Also, then the only thing you need to do during build is inline the symbol file.

**Comment by Munter** [comment]
The development version you specify will not let you apply that styling as far as I understand. The aim of the builder is to create a similar but optimized version. What you seem to be proposing is a new and potentially breaking behavior

**Comment by rolandliwag** [comment]
@slang800 thanks for that link; it looks useful.

@Munter No, it wouldn't work as-is. Here's what we're doing to make this work right now:
```
<a data-bind="html: replaceWithSvg(GETSTATICURL('/path/to/icon.svg'))"></a> 
```

Where ```replaceWithSvg``` makes a request to the URL and replaces the tag's innerHtml with the svg content. It would be nice to get rid of that.

**Comment by Munter** [comment]
It's not hard to follow relations to an svg and manipulate the html to contain an inline version of it. The hard part is figuring our a good development environment solution where it still works with the svg as an external file. One which is generic enough to be used across frameworksand distinctive enough to be picked up by static analysis. I don't now of any working solution where you are able to apply css from a html document to an svg that is an external file, so I don't see a good development solution for this.

You can obviously make your own transform, but I would be against making it a part of the builder if it requires a very specific set of locally set up tools and frameworks

**Comment by silvenon** [comment]
You can also check out [Grunticon](http://www.grunticon.com/), it has exactly this behavior, where you can add `data-grunticon-embed` to tell Grunticon loader to embed the SVG both in development and production.

I suggest using `<symbol>` (+ [svg4everybody](https://github.com/jonathantneal/svg4everybody)) most of the times because it takes less space and is less hacky, but for those few cases where you need to heavily style the SVG (or modify than 2 colors) you can use Grunticon.

Here's an example of using `<symbol>`s:

`images/icons.svg`
```xml
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
	<symbol id="heart" viewBox="0 0 30 25">
		<title>Heart</title>
		<path stroke-miterlimit="10" d="M13.9,22.9l1,1.1l1-1.1c1.8-1.7,4.6-4.2,8.4-6.8
			C31.3,11.7,30.2,1,22.4,1c-4.2,0-6.3,2.2-7.4,3.9C13.9,3.2,11.8,1,7.6,1c-7.9,0-8.9,10.7-2.1,15.2C9.3,18.7,12.2,21.2,13.9,22.9z"/>
	</symbol>
</svg>
```

`index.html`
```html
<svg class="icon" title="Heart" width="30" height="25">
	<use xlink:href="images/icons.svg#heart"></use>
</svg>
```

`main.css`
```css
.icon {
	fill: red;
	stroke: black;
}
```

Make sure not to set any attributes in `images/icons.svg` that you plan to modify via CSS, otherwise you will not be able to override that attribute because `<use>` inserts content of the `<symbol>` via some Shadow DOM thingy.

Whew, with this I conclude my intro to SVG icons :smiley:

**Comment by rolandliwag** [comment]
Thanks a bunch! That's actually super helpful. I think I actually have enough here to achieve what I want but not exactly in the way that I thought it should have been done (perhaps a better way). I'll close this now.
